### ISSUE TITLE
Change in `find_sexp_type`?

### ISSUE THREAD

**Issue by drisso** [tatami-inc/beachmat#17]
Hi Aaron,

I just noticed that our `mbkmeans` package is failing in bioc-devel, possibly due to changes to beachmat. The actual code of our `mini_batch()` function has not changed, but while it works with the release version of beachmat, it gives the following error in the devel version:

``` 
Error in mini_batch(data = t(x), clusters = clusters, batch_size = batch_size,  : 
  Cannot convert object to a function: [type=NULL; target=CLOSXP, SPECIALSXP, or BUILTINSXP].
```

This errors happens only when `x` is a `DelayedMatrix` (HDF5-backed), while all works if `x` is a numeric matrix. 

The line that seems to be the culprit is this:
```
auto matrix_type=beachmat::find_sexp_type(data);
```

Anything I'm doing wrong? Or any idea of what's going on?

**Comment by LTLA** [comment]
Fixed yesterday in 40ce1f32858dfa8e462679e0ae5e15dab5e94c99.

**Comment by drisso** [comment]
Oh, awesome! Thanks!
